board/safety_declarations.h

```diff
@@ -208,39 +212,39 @@ void pcm_cruise_check(bool cruise_engaged);
 void safety_tick(const safety_config *safety_config);
 
 // This can be set by the safety hooks
-bool controls_allowed = false;
-bool relay_malfunction = false;
-bool gas_pressed = false;
-bool gas_pressed_prev = false;
-bool brake_pressed = false;
-bool brake_pressed_prev = false;
-bool regen_braking = false;
-bool regen_braking_prev = false;
-bool cruise_engaged_prev = false;
-struct sample_t vehicle_speed;
-bool vehicle_moving = false;
-bool acc_main_on = false;  // referred to as "ACC off" in ISO 15622:2018
-int cruise_button_prev = 0;
-bool safety_rx_checks_invalid = false;
+extern bool controls_allowed;
```

Comment:
why do we need to specify extern?

Reply:
`controls_allowed` is a global variable that is declared in safety_declarations.h and defined with external linkage in  safety.h

Misra 8.4 says that when defining with external linkage, you must have an equivalent declaration somewhere (so including the `extern` keyword)